File: backend/src/main/java/com/backend/ord/services/UserService.java
package com.backend.ord.services;

import com.backend.ord.domain.entities.User;

import java.util.List;
import java.util.UUID;
import java.util.Optional;
public interface UserService {
    List<User> getAll();

    Optional<User> findById(UUID id);

    User save(User user);

    Optional<User> findUserByAuthToken(String authToken);
}

File: backend/src/main/kotlin/com/backend/ord/services/UserService.kt
package com.backend.ord.services

import com.backend.ord.domain.entities.User
import java.util.*

interface UserService {
    fun findAll(): List<User>

    fun findById(id: UUID): User?

    fun save(user: User): User?

    fun findUserByAuthToken(authToken: String): User?

    fun findUserByEmail(email: String): User?
}
